src/EncryptionUtilsV2.cs

```diff
@@ -337,7 +337,7 @@ private static PutObjectRequest GetInstructionFileRequest(string bucketName, str
         /// <param name="instructions">
         /// The instruction that will be used to encrypt the object data.
         /// </param>
-        internal static void DecryptObjectUsingInstructionsV2(GetObjectResponse response, EncryptionInstructions instructions)
+        internal static void DecryptObjectUsingInstructionsGcm(GetObjectResponse response, EncryptionInstructions instructions)
```

Comment:
The `V2` moniker is not descriptive in this context. The method is decrypting using GCM. This was supported in `V1` as well, so referring to the GCM implementation as `V2` is not accurate. It is a common pitfall to think that S3 EC `V2` introduced GCM, but it only made GCM the default. 

That said, your point about convention is valid. Perhaps the right thing to do is to move the method to `EncryptionUtils` since that seems to include stuff from both `V1` and `V2`, and doesn't include suffixes.

Reply:
Concerning the naming conventions, I will take another look and decide

Reply:
Okay on the renaming. I agree `Gcm` is more descriptive, and this is `internal` so it's a two-way door.

I'm okay if we merge this PR to a feature branch and add tests in a follow-up PR. But generally we should not release new code without tests, otherwise a failed test won't prevent a potentially broken feature from being released. 

In my opinion:
* tests should be "atomic" with the code under test - once they're separate backlog items prioritization becomes challenging.
* Especially when fixing a big that existing tests didn't cover, PRs should introduce automated testing to protect against regressions in the future.

That said, I do think we want a end-to-end integration test that goes through the SES worfklow. Is it possible to either mock the response from S3 without the metadata field, or else prepare an S3 object directly without going through SES?

Reply:
Writing an end-to-end test can be tricky because in order to reproduce the bug:

- We need a dev email domain to be able to setup AWS SES for testing purposes

- We need to send an email to this domain. I did it manually by using AWS SES user interface (I am not sure how it w works behind the scenes and whether we can just send a regular email the dev email domain without using the UI)

The best way to test this bug is to have an already encrypted S3 object by AWS SES and decrypt it using our library. In order to achieve this, we also need to use the same AWS KMS key that was used for encrypting the object.